haos_extend.c does not build with recent versions of OpenIPMI

trying to build simengine on Fedora34 (and later Centos8 / RHEL8):

OpenIPMI-devel-2.0.31

+ gcc -shared -o /home/fabbione/rpmbuild/BUILD/simengine-3.36/haos_extend.so -fPIC /home/fabbione/rpmbuild/BUILD/simengine-3.36/enginecore/ipmi_sim/haos_extend.c
/home/fabbione/rpmbuild/BUILD/simengine-3.36/enginecore/ipmi_sim/haos_extend.c: In function 'bmc_set_chassis_control':
/home/fabbione/rpmbuild/BUILD/simengine-3.36/enginecore/ipmi_sim/haos_extend.c:118:8: error: 'CHASSIS_CONTROL_POWER' undeclared (first use in this function); did you mean 'IPMI_CONTROL_POWER'?
  118 |   case CHASSIS_CONTROL_POWER:
      |        ^~~~~~~~~~~~~~~~~~~~~
      |        IPMI_CONTROL_POWER
/home/fabbione/rpmbuild/BUILD/simengine-3.36/enginecore/ipmi_sim/haos_extend.c:118:8: note: each undeclared identifier is reported only once for each function it appears in
/home/fabbione/rpmbuild/BUILD/simengine-3.36/enginecore/ipmi_sim/haos_extend.c:119:8: error: 'CHASSIS_CONTROL_RESET' undeclared (first use in this function); did you mean 'IPMI_CONTROL_RESET'?
  119 |   case CHASSIS_CONTROL_RESET:
      |        ^~~~~~~~~~~~~~~~~~~~~
      |        IPMI_CONTROL_RESET
/home/fabbione/rpmbuild/BUILD/simengine-3.36/enginecore/ipmi_sim/haos_extend.c:120:8: error: 'CHASSIS_CONTROL_BOOT_INFO_ACK' undeclared (first use in this function)
  120 |   case CHASSIS_CONTROL_BOOT_INFO_ACK:
      |        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/fabbione/rpmbuild/BUILD/simengine-3.36/enginecore/ipmi_sim/haos_extend.c:121:8: error: 'CHASSIS_CONTROL_BOOT' undeclared (first use in this function)
  121 |   case CHASSIS_CONTROL_BOOT:
      |        ^~~~~~~~~~~~~~~~~~~~
/home/fabbione/rpmbuild/BUILD/simengine-3.36/enginecore/ipmi_sim/haos_extend.c:122:8: error: 'CHASSIS_CONTROL_GRACEFUL_SHUTDOWN' undeclared (first use in this function); did you mean 'HW_OP_CAN_GRACEFUL_SHUTDOWN'?
  122 |   case CHASSIS_CONTROL_GRACEFUL_SHUTDOWN:
      |        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |        HW_OP_CAN_GRACEFUL_SHUTDOWN
/home/fabbione/rpmbuild/BUILD/simengine-3.36/enginecore/ipmi_sim/haos_extend.c: In function 'ipmi_sim_module_init':
/home/fabbione/rpmbuild/BUILD/simengine-3.36/enginecore/ipmi_sim/haos_extend.c:179:8: warning: implicit declaration of function 'ipmi_mc_alloc_unconfigured' [-Wimplicit-function-declaration]
  179 |   rv = ipmi_mc_alloc_unconfigured(sys, 0x20, &bmc_mc);
      |        ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/fabbione/rpmbuild/BUILD/simengine-3.36/enginecore/ipmi_sim/haos_extend.c:188:3: warning: implicit declaration of function 'ipmi_mc_set_chassis_control_func'; did you mean 'bmc_set_chassis_control'? [-Wimplicit-function-declaration]
  188 |   ipmi_mc_set_chassis_control_func(bmc_mc, bmc_set_chassis_control, bmc_get_chassis_control, sys);
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |   bmc_set_chassis_control
error: Bad exit status from /var/tmp/rpm-tmp.opZSSk (%build)


RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.opZSSk (%build)

--- a/enginecore/ipmi_sim/haos_extend.c
+++ b/enginecore/ipmi_sim/haos_extend.c
@@ -23,7 +23,12 @@
 
 #include <OpenIPMI/ipmi_msgbits.h>
 #include <OpenIPMI/ipmi_bits.h>
-#include <OpenIPMI/serv.h>
+
+#ifdef INCLUDE_MCSERV_H
+  #include <OpenIPMI/mcserv.h>
+#else
+  #include <OpenIPMI/serv.h>
+#endif
 
 #define PVERSION "0.0.0"
 

--- a/enginecore/ipmi_sim/haos_extend.c
+++ b/enginecore/ipmi_sim/haos_extend.c
@@ -24,7 +24,7 @@
 #include <OpenIPMI/ipmi_msgbits.h>
 #include <OpenIPMI/ipmi_bits.h>
 
-#ifdef INCLUDE_MCSERV_H
+#ifdef OPENIPMI_POST_2_0_30
   #include <OpenIPMI/mcserv.h>
 #else
   #include <OpenIPMI/serv.h>
@@ -181,7 +181,11 @@ int ipmi_sim_module_init(sys_data_t *sys, const char *options)
 
   free(initstr);
 
+#ifdef OPENIPMI_POST_2_0_30
+  rv = is_mc_alloc_unconfigured(sys, 0x20, &bmc_mc);
+#else
   rv = ipmi_mc_alloc_unconfigured(sys, 0x20, &bmc_mc);
+#endif
 
   if (rv)
   {

--- a/rpm/specfiles/simengine-core.spec
+++ b/rpm/specfiles/simengine-core.spec
@@ -7,12 +7,11 @@ License:   GPLv3+
 
 %global gittag %{version}
 %global selected_libdir /usr/lib64
-%global openipmi_version 2.0.28
 
 Source0: https://github.com/Seneca-CDOT/simengine/archive/%{gittag}/simengine-%{version}.tar.gz  
 
-BuildRequires: OpenIPMI-devel = %{openipmi_version}, gcc
-Requires: simengine-database, python3-libvirt, OpenIPMI = %{openipmi_version}, OpenIPMI-lanserv = %{openipmi_version}, python3-redis, python3-pysnmp, python3-neo4j-driver, python3-websocket-client
+BuildRequires: OpenIPMI-devel, gcc
+Requires: simengine-database, python3-libvirt, OpenIPMI, OpenIPMI-lanserv, python3-redis, python3-pysnmp, python3-neo4j-driver, python3-websocket-client
 
 %description
 Core files for SimEngine.

--- a/rpm/specfiles/simengine-core.spec
+++ b/rpm/specfiles/simengine-core.spec
@@ -26,12 +26,12 @@ Core files for SimEngine.
 
 %build
 openipmi_version=$(rpm -q --qf "%%{VERSION}" OpenIPMI-devel)
-define_include_mcserv_h=$([[ "$openipmi_version" > "2.0.30" ]] && printf "%s" "-D INCLUDE_MCSERV_H")
+define_openipmi_post_2_0_30=$([[ "$openipmi_version" > "2.0.30" ]] && printf "%s" "-D OPENIPMI_POST_2_0_30")
 gcc \
     -shared \
     -o %{_builddir}/simengine-%{version}/haos_extend.so \
     -fPIC \
-    $define_include_mcserv_h \
+    $define_openipmi_post_2_0_30 \
     %{_builddir}/simengine-%{version}/enginecore/ipmi_sim/haos_extend.c
 
 %install

--- a/rpm/specfiles/simengine-core.spec
+++ b/rpm/specfiles/simengine-core.spec
@@ -25,7 +25,14 @@ Core files for SimEngine.
 %autosetup -n simengine-%{version}
 
 %build
-gcc -shared -o %{_builddir}/simengine-%{version}/haos_extend.so -fPIC %{_builddir}/simengine-%{version}/enginecore/ipmi_sim/haos_extend.c
+openipmi_version=$(rpm -q --qf "%%{VERSION}" OpenIPMI-devel)
+define_include_mcserv_h=$([[ "$openipmi_version" > "2.0.30" ]] && printf "%s" "-D INCLUDE_MCSERV_H")
+gcc \
+    -shared \
+    -o %{_builddir}/simengine-%{version}/haos_extend.so \
+    -fPIC \
+    $define_include_mcserv_h \
+    %{_builddir}/simengine-%{version}/enginecore/ipmi_sim/haos_extend.c
 
 %install
 mkdir -p %{buildroot}%{_datadir}/simengine/

